planner: max_min_eliminate panics on empty scalar aggregation produced by SEMI_JOIN_REWRITE

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

A deterministic reproducer at the optimizer-rule seam is:

```go
sctx := coretestsdk.MockContext()
agg := logicalop.LogicalAggregation{}.Init(sctx, 0)
_, _, _ = (&rule.MaxMinEliminator{}).Optimize(context.Background(), agg)
```

On current master without the guard below, the call panics with:

```text
panic: runtime error: index out of range [0] with length 0
```

This empty scalar aggregation shape is expected to be reachable from `SEMI_JOIN_REWRITE` on `EXISTS` subqueries, for example queries shaped like:

```sql
select *
from t
where exists (
  select /*+ SEMI_JOIN_REWRITE() */ 1
  from t as x
  where x.a < 1
);
```

### 2. What did you expect to see? (Required)

`max_min_eliminate` should treat an empty scalar aggregation as non-applicable input and return it unchanged, without panicking.

### 3. What did you see instead (Required)

`composeAggsByInnerJoin()` assumes `aggs[0]` exists. When `LogicalAggregation` has `AggFuncs=[]` and `GroupByItems=[]`, `splitAggFuncAndCheckIndices()` returns `(empty slice, true)`, and `max_min_eliminate` panics with `index out of range [0] with length 0`.

### 4. What is your TiDB version? (Required)

Current local master (branch `fix/max-min-eliminate-empty-aggs-panic`, based on a worktree that was `master...origin/master [ahead 10706]` before branching).

--- a/pkg/planner/core/rule/rule_max_min_eliminate.go
+++ b/pkg/planner/core/rule/rule_max_min_eliminate.go
@@ -226,6 +226,9 @@ func (a *MaxMinEliminator) eliminateMaxMin(p base.LogicalPlan) base.LogicalPlan 
 		if len(agg.GroupByItems) != 0 {
 			return agg
 		}
+		if len(agg.AggFuncs) == 0 {
+			return agg
+		}
 		// Make sure that all of the aggFuncs are Max or Min.
 		for _, aggFunc := range agg.AggFuncs {
 			if aggFunc.Name != ast.AggFuncMax && aggFunc.Name != ast.AggFuncMin {

--- a/pkg/planner/core/rule/rule_max_min_eliminate_test.go
+++ b/pkg/planner/core/rule/rule_max_min_eliminate_test.go
@@ -1,0 +1,37 @@
+// Copyright 2026 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rule
+
+import (
+	"context"
+	"testing"
+
+	"github.com/pingcap/tidb/pkg/planner/core/operator/logicalop"
+	"github.com/pingcap/tidb/pkg/planner/util/coretestsdk"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMaxMinEliminateSkipsEmptyScalarAgg(t *testing.T) {
+	sctx := coretestsdk.MockContext()
+	agg := logicalop.LogicalAggregation{}.Init(sctx, 0)
+	opt := &MaxMinEliminator{}
+
+	require.NotPanics(t, func() {
+		p, changed, err := opt.Optimize(context.Background(), agg)
+		require.NoError(t, err)
+		require.False(t, changed)
+		require.Same(t, agg, p)
+	})
+}

--- a/tests/integrationtest/r/planner/core/casetest/integration.result
+++ b/tests/integrationtest/r/planner/core/casetest/integration.result
@@ -509,6 +509,26 @@ HashJoin	root		inner join, equal:[eq(planner__core__casetest__integration.t.a, p
 │     └─TableFullScan	cop[tikv]	table:t1	keep order:false, stats:pseudo
 └─TableReader(Probe)	root		data:TableFullScan
   └─TableFullScan	cop[tikv]	table:t	keep order:false, stats:pseudo
+drop table if exists t;
+create table t(a int);
+insert into t values (0), (1);
+set @@tidb_opt_enable_semi_join_rewrite = on;
+with c as (select * from t)
+select max(t.a)
+from t
+where exists (
+select /*+ SEMI_JOIN_REWRITE() */ 1
+from c
+where c.a < 1
+)
+and exists (
+select 1
+from c
+where c.a < 2
+);
+max(t.a)
+1
+set @@tidb_opt_enable_semi_join_rewrite = default;
 drop table if exists test;
 create table test(id int, value int);
 drop table if exists t;

--- a/tests/integrationtest/t/planner/core/casetest/integration.test
+++ b/tests/integrationtest/t/planner/core/casetest/integration.test
@@ -136,6 +136,26 @@ explain format='plan_tree' select * from t where exists (select /*+ SEMI_JOIN_RE
 explain format='plan_tree' select /*+ hash_join_build(t) */ * from t where exists (select /*+ SEMI_JOIN_REWRITE() */ 1 from t t1 join t t2 where t1.a = t2.a and t1.a = t.a);
 explain format='plan_tree' select /*+ hash_join_probe(t) */ * from t where exists (select /*+ SEMI_JOIN_REWRITE() */ 1 from t t1 join t t2 where t1.a = t2.a and t1.a = t.a);
 
+# TestSemiJoinRewriteWithCTEAndMultipleExists
+drop table if exists t;
+create table t(a int);
+insert into t values (0), (1);
+set @@tidb_opt_enable_semi_join_rewrite = on;
+with c as (select * from t)
+select max(t.a)
+from t
+where exists (
+  select /*+ SEMI_JOIN_REWRITE() */ 1
+  from c
+  where c.a < 1
+)
+and exists (
+  select 1
+  from c
+  where c.a < 2
+);
+set @@tidb_opt_enable_semi_join_rewrite = default;
+
 # TestDecorrelateLimitInSubquery
 drop table if exists test;
 create table test(id int, value int);

